snap/get_version

```diff
@@ -0,0 +1,114 @@
+#!/bin/dash
+set -eu
+
+# Script to build the version to set to the snap.
+
+# When considering a tag, if starting with "<current branch name>-" it will have its prefix removed.
+# For instance:
+# * 0.1 -> 0.1
+# * msentraid-0.1 -> 0.1 on msentraid branch.
+
+# 1. If current commit is tagged, the version is directly the tag name.
+# 2. If current commit is not tagged, the version is:
+#    a. <last tag name on current branch>+<commit_sha> for main branch.
+#    b. <last tag name on current branch (not in main)>+<commit_sha>.<last_commit_merged_from_main> for other branches.
+#
+# Any of those version will be annoted with +dirty if there are local changes.
+
+# set_version will markup the version in the snapcraft.yaml file after amending it with a dirty markup if necessary.
+# $1: version: the version to set.
+set_version() {
+    version="${1}"
+
+    version=$(annotate_with_dirty "${version}")
+    craftctl set version="${version}"
+}
+
+# annotate_with_dirty may amend the version with a dirty markup if there are local changes.
+# $1: version: the version to annotate.
+annotate_with_dirty() {
+    version="${1}"
+
+    # check if current tree content is dirty.
+    if [ -n "$(git -C ${SNAPCRAFT_PART_SRC} status --porcelain)" ]; then
```

Comment:
All occurrences of `${SNAPCRAFT_PART_SRC} ` in this file need to be double-quoted to prevent globbing and word splitting

```suggestion
    if [ -n "$(git -C "${SNAPCRAFT_PART_SRC}" status --porcelain)" ]; then
```

Reply:
This closes the older double quote if I am correct and reopen it after ${SNAPCRAFT_PART_SRC}?

Reply:
That was on the outdated code. The new one had the flaw too, I fixed it. Thanks.